packages/ui/src/Toggle.module.scss

```diff
@@ -0,0 +1,98 @@
+@use '../styles/constants' as c;
+@use '../styles/helpers' as h;
+
+$h: 24px;
+$w: 60px;
+$knob-r: $w / 2 + 8px;
+$track-br: 22px;
+$track-bg-on: #1dbc22;
+$track-bg-off: c.$colorNeutralLight; // rgba(0,0,0,.32);
+$knob-bg-on: c.$colorNeutralWhite;
+$knob-bg-off: c.$colorNeutralLighter;
+$label-margin: 12px;
+$knob-h: 20px;
+$knob-w: 20px;
+$knob-offset: 2px;
+
+@keyframes fadein {
+  from {
+    opacity: 0;
+  }
+  to {
+    opacity: 1;
+  }
+}
+
+.wrapper {
+  // hide the actual input, we will show the toggle-track instead.
+  input[type='checkbox'] {
+    display: none;
+  }
+
+  // faux wrapping label
+  label {
+    cursor: pointer;
+    display: inline-flex; // for aligning with the .labelText
+    align-items: center;
+  }
+
+  // actual text label
+  .labelText {
+    margin-right: $label-margin;
+  }
+
+  // the `track` (default off state)
+  .toggle-track {
+    display: inline-block;
+    height: $h;
+    width: $w;
+    background: $track-bg-off;
+    border-radius: $track-br;
+    position: relative;
+  }
+
+  // track on state
+  input[type='checkbox']:checked + label .toggle-track {
+    background: $track-bg-on;
+  }
+
+  // the `knob` (default off state)
+  .toggle-track:before {
+    content: '';
+    display: inline-block;
+    height: $knob-h;
+    width: $knob-w;
+    background: $knob-bg-off;
+    border-radius: 50%;
+    position: absolute;
+    top: $knob-offset;
+    right: $knob-offset;
+    transition: right 0.2s ease-in;
+  }
+
+  // knob on state
+  input[type='checkbox']:checked + label .toggle-track:before {
+    background: $knob-bg-on;
+    right: $knob-r;
+  }
+
+  // th "ON" text
+  input[type='checkbox']:checked + label .toggle-track:after {
+    content: 'ON';
+    position: absolute;
+    top: 2px;
+    right: 8px;
+    font-size: 0.9rem;
+    color: #fff;
+    animation: fadein 0.2s;
```

Comment:
Should we create `styles/constants/_animations.scss` and add `0.2s` as `$transitionTimeMedium` there?

Reply:
done.